superset/connectors/sqla/models.py

```diff
@@ -229,6 +230,30 @@ class TableColumn(Model, BaseColumn, CertificationMixin):
     update_from_object_fields = [s for s in export_fields if s not in ("table_id",)]
     export_parent = "table"
 
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """
+        Construct a TableColumn object.
+
+        Historically a TableColumn object (from an ORM perspective) was tighly bound to
+        a SqlaTable object, however with the introduction of the Query datasource this
+        is no longer true, i.e., the SqlaTable relationship is optional.
+
+        Now the TableColumn is either directly associated with the Database object (
+        which is unkwnonw to the ORM) or indirectly via the SqlaTable object (courtesy
+        of the ORM) depending on the context.
+        """
+
+        self._database = kwargs.pop("database", None)
```

Comment:
Given that the `database` variable isn't part of the ORM I believe I'm correct in saying that it can only be defined via a keyword argument.

Reply:
@john-bodley is there a follow up PR where'd you be using the `database` arg? also can you annotate this somehow to let people know this is going to be an ORM object vs some might assume it's an id

Reply:
@hughhhh it's used in this PR [here](https://github.com/apache/superset/pull/24121/files#diff-94d3bd02afcbd1983033ed9c33e9efb466cb9a64e3b1f3954e6e0d627f70d0efR196).

I've added a type annotation for clarity.